File: internal/room/domain/room.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Room represents a physical room (classroom, lab, etc.) within a tenant.
type Room struct {
	ID        uuid.UUID
	Name      string
	Code      string // unique identifier e.g. "A101"
	Building  string
	Floor     int
	Capacity  int
	Equipment []string
	IsActive  bool
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: web/packages/api-client/src/types/room.ts
// Room module types matching backend room/availability handlers.

export interface Room {
  id: string;
  name: string;
  code: string;
  building: string;
  floor: number;
  capacity: number;
  equipment: string[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateRoomRequest {
  name: string;
  code: string;
  building: string;
  floor: number;
  capacity: number;
  equipment: string[];
}

export interface UpdateRoomRequest {
  name: string;
  code: string;
  building: string;
  floor: number;
  capacity: number;
  equipment: string[];
  is_active: boolean;
}

export interface RoomFilter {
  building?: string;
  min_capacity?: number;
  equipment?: string;
}

export interface RoomAvailabilitySlot {
  day: number;
  period: number;
  is_available: boolean;
}
